web/src/lib/utils/asset-utils.ts

```diff
@@ -224,6 +227,44 @@ export const getSelectedAssets = (assets: Set<AssetResponseDto>, user: UserRespo
   return ids;
 };
 
+export const selectAllAssets = async (assetStore: AssetStore, assetInteractionStore: AssetInteractionStore) => {
```

Comment:
Maybe we can change `assetInteractionStore.selectAsset` to accept an array of assets and omit the second loop here altogether

Reply:
The second loop is useful to allow the UI to be refreshed during the "select all" process, with `delay(0)`

Reply:
I just committed that change, but to be honest there is no perceptible performance improvement. Anyway that's still cleaner